core/src/main/java/com/linecorp/armeria/client/retry/ExponentialBackoffBuilder.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.retry;
+
+/**
+ * A builder for creating instances of {@link ExponentialBackoff}.
+ *
+ * <p>This builder allows you to configure an exponential backoff strategy by specifying
+ * the initial delay, the maximum delay, and a multiplier. The exponential backoff
+ * increases the delay between retries exponentially, starting from the initial delay and
+ * multiplying the delay by the specified multiplier after each retry, up to the maximum delay.</p>
+ *
+ * <p>Example usage:</p>
+ *
+ * <pre>
+ * {@code
+ * ExponentialBackoff backoff = new ExponentialBackoffBuilder()
+ *     .initialDelayMillis(100)
+ *     .maxDelayMillis(10000)
+ *     .multiplier(2.0)
+ *     .build();
+ * }
+ * </pre>
+ *
+ * @see ExponentialBackoff
+ */
+public class ExponentialBackoffBuilder {
+    private long initialDelayMillis;
+    private long maxDelayMillis;
+    private double multiplier;
```

Comment:
Should we follow the default values used in the factory methods?
```suggestion
    private double multiplier = 2.0;
```